tests/e2e/specs/product-editor/block-integration.test.js

```diff
@@ -0,0 +1,693 @@
+		await editorUtils.fillProductName();
+		await editorUtils.clickPluginTab();
+
+		const {
+			gtin,
+			mpn,
+			brand,
+			condition,
+			gender,
+			size,
+			sizeSystem,
+			sizeType,
+			color,
+			material,
+			pattern,
+			ageGroup,
+			multipack,
+			isBundle,
+			availabilityDate,
+			availabilityTime,
+			adultContent,
+		} = editorUtils.getAllProductAttributes();
+
+		const pairs = [
+			[ gtin, '3234567890126' ],
+			[ mpn, 'GO12345OOGLE' ],
+			[ brand, 'e2e_test_woocommerce_brands' ],
+			[ condition, 'new' ],
+			[ gender, 'unisex' ],
+			[ size, 'Good for everybody' ],
+			[ sizeSystem, 'JP' ],
+			[ sizeType, 'regular' ],
+			[ color, 'Cherry blossom' ],
+			[ material, 'Titanium alloy' ],
+			[ pattern, 'Cyberpunk' ],
+			[ ageGroup, 'kids' ],
+			[ multipack, '9999' ],
+			[ isBundle, 'no' ],
+			[ availabilityDate, '2024-02-29' ],
+			[ availabilityTime, '23:59' ],
+			[ adultContent, 'no' ],
+		];
+
+		/*
+		 * Assert:
+		 * - All attributes are empty or default
+		 * - Save all attributes
+		 * - After saving, attribute values remain the same
+		 */
+		for ( const [ attribute, value ] of pairs ) {
+			await expect( attribute ).toHaveValue( '' );
+			await editorUtils.setAttributeValue( attribute, value );
+		}
+
+		await editorUtils.save();
+
+		for ( const [ attribute, value ] of pairs ) {
+			await expect( attribute ).toHaveValue( value );
+		}
+
+		/*
+		 * Assert:
+		 * - It allows to save all attributes to empty or default
+		 */
+		for ( const [ attribute ] of pairs ) {
+			await editorUtils.setAttributeValue( attribute, '' );
+		}
+
+		await editorUtils.save();
+
+		for ( const [ attribute ] of pairs ) {
+			await expect( attribute ).toHaveValue( '' );
+		}
+	} );
+
+	test( 'Save product attributes to variation product', async () => {
+		await editorUtils.gotoEditVariableProductPage();
+		await editorUtils.gotoEditVariationProductPage();
+		await editorUtils.clickPluginTab();
+
+		/*
+		 * Since the testing of all attributes is already covered in the simple product test case above.
+		 * This case only tests a few key attributes to ensure the data saving works for variation product.
+		 */
+		const {
+			condition,
+			color,
+			multipack,
+			availabilityDate,
+			availabilityTime,
+		} = editorUtils.getAllProductAttributes();
+
+		const pairs = [
+			[ condition, 'new' ],
+			[ color, 'Cherry blossom' ],
+			[ multipack, '9999' ],
+			[ availabilityDate, '2024-02-29' ],
+			[ availabilityTime, '23:59' ],
+		];
+
+		for ( const [ attribute ] of pairs ) {
+			await expect( attribute ).toHaveValue( '' );
+		}
+
+		for ( const [ attribute, value ] of pairs ) {
+			await expect( attribute ).toHaveValue( '' );
+			await editorUtils.setAttributeValue( attribute, value );
+		}
+
+		await editorUtils.save();
+
+		for ( const [ attribute, value ] of pairs ) {
+			await expect( attribute ).toHaveValue( value );
+		}
+	} );
+
+	test.afterEach( async () => {
+		await page.unrouteAll();
```

Comment:
Why we need this?

Reply:
This is to removes all routes created with page.route() via the `editorUtils.mockChannelVisibility` calls in the "Change channel visibility and check its notice, status, and issues" case. In this way, the side effects between test cases can be reduced.